autowarefoundation/autoware.universe
# feat(rtc_auto_approver): add rtc_auto_approver

Signed-off-by: Fumiya Watanabe <rej55.g@gmail.com>

## Description

To approve the request to cooperate from behavior planning modules, I add  `rtc_auto_approver`.
`rtc_auto_approver` approves the request if the corresponding module outputs safe status.



## Related links

https://github.com/autowarefoundation/autoware.universe/issues/891


## Tests performed

I checked by the planning simulator with the following PR.

https://github.com/autowarefoundation/autoware.universe/issues/891

1. run planning simulator
2. set the route including some regulatory elements (e.g. crosswalk or intersection)
3. check that the behavior velocity planner modules set some stop points
4. launch rtc_auto_approver
5. check that the stop points are removed
6. check that the stop points are set if the modules output unsafe status.


## Notes for reviewers

Adding launch configuration to launch this node is next action.


## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
